eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/ZmqWatcher.scala

```diff
@@ -226,13 +238,128 @@ class ZmqWatcher(chainHash: ByteVector32, blockCount: AtomicLong, client: Extend
+            Future.failed(t)
+        }
+      case PublishStrategy.JustPublish =>
+        log.info("publishing tx: txid={} tx={}", p.tx.txid, p.tx)
+        publish(p.tx, isRetry = false)
+    }
+  }
+
+  def publish(tx: Transaction, isRetry: Boolean): Future[ByteVector32] = {
+    client.publishTransaction(tx)(singleThreadExecutionContext).recoverWith {
       case t: Throwable if t.getMessage.contains("(code: -25)") && !isRetry => // we retry only once
         import akka.pattern.after
-        after(3 seconds, context.system.scheduler)(Future.successful({})).map(_ => publish(tx, isRetry = true))
-      case t: Throwable => log.error("cannot publish tx: reason={} txid={} tx={}", t.getMessage, tx.txid, tx)
+        after(3 seconds, context.system.scheduler)(Future.successful({})).flatMap(_ => publish(tx, isRetry = true))
+      case t: Throwable =>
+        log.error("cannot publish tx: reason={} txid={}", t.getMessage, tx.txid)
+        Future.failed(t)
+    }
+  }
+
+  def publishCommitWithAnchor(commitTx: Transaction, currentFeerate: FeeratePerKw, targetFeerate: FeeratePerKw, dustLimit: Satoshi, signingKit: TransactionSigningKit.ClaimAnchorOutputSigningKit): Future[ByteVector32] = {
+    import signingKit._
+    if (targetFeerate <= currentFeerate) {
+      log.info(s"publishing commit tx without the anchor (current feerate=$currentFeerate): txid=${commitTx.txid}")
+      publish(commitTx, isRetry = false)
+    } else {
+      log.info(s"publishing commit tx with the anchor (target feerate=$targetFeerate): txid=${commitTx.txid}")
+      // We want the feerate of the package (commit tx + tx spending anchor) to equal targetFeerate.
+      // Thus we have: anchorFeerate = targetFeerate + (weight-commit-tx / weight-anchor-tx) * (targetFeerate - commitTxFeerate)
+      // If we use the smallest weight possible for the anchor tx, the feerate we use will thus be greater than what we want,
+      // and we can adjust it afterwards by raising the change output amount.
+      val anchorFeerate = targetFeerate + FeeratePerKw(targetFeerate.feerate - currentFeerate.feerate) * commitTx.weight() / Transactions.claimAnchorOutputMinWeight
+      // NB: bitcoind requires txs to have at least one output, but we'll remove it later to keep a single change output.
+      // In case we have the perfect set of utxo amounts and no change output is added, we need the amount to be greater
+      // than the fee because we may need to deduce the fee from that output.
+      val dummyChangeAmount = Transactions.weight2fee(anchorFeerate, Transactions.claimAnchorOutputMinWeight) + dustLimit
+      publish(commitTx, isRetry = false).flatMap(commitTxId => {
+        val txNotFunded = Transaction(2, Nil, TxOut(dummyChangeAmount, Script.pay2wpkh(Transactions.PlaceHolderPubKey)) :: Nil, 0)
+        client.fundTransaction(txNotFunded, FundTransactionOptions(anchorFeerate))(singleThreadExecutionContext)
```

Comment:
Damn, that's going to be painful...

Reply:
Actually it's not that painful because I should not need to unlock these utxos :sweat_smile: 

- Either `fundrawtransaction` succeeded and locked utxos and in that case there's no reason `signrawtransaction`/`publish` would fail, so we're good.
- Or `fundrawtransaction` failed and utxos are not locked.

For extra safety later when we move to a model where we have dedicated fee bumper actors and have (hopefully) simpler code I'll add code to cover that case, but it should never happen, so I think we're ok for this PR.

Reply:
Done in https://github.com/ACINQ/eclair/pull/1702/commits/090f7cba3b80eb4335eab9716b7b2154db5c4ef3

Reply:
Indeed, it was the same for the funding tx. The only catch is we can keep utxos locked if we stop eclair at that same time, but it's not really a big deal.